File: scripts/update-social-media.py
#!/usr/bin/env python3
"""
Script to update coach social media links with random combinations
"""

import psycopg2
from psycopg2.extras import RealDictCursor
import ssl

DATABASE_URL = "postgresql://postgres.vosbykmmrfcyrdrkvinx:Felixisagenius!@aws-1-ap-southeast-2.pooler.supabase.com:5432/postgres"

# Random social media combinations
social_combos = [
    {"fb": "https://facebook.com/coach.swimming.hk", "ig": None, "web": None},
    {"fb": None, "ig": "https://instagram.com/yoga_with_maggie", "web": None},
    {"fb": None, "ig": None, "web": "https://coachchan.com"},
    {"fb": "https://facebook.com/basketballcoachhk", "ig": "https://instagram.com/hkbasketballcoach", "web": None},
    {"fb": "https://facebook.com/tenniscoach.hk", "ig": None, "web": "https://tenniscoachhk.com"},
    {"fb": None, "ig": "https://instagram.com/pilateslife.hk", "web": "https://pilateshk.com"},
    {"fb": "https://facebook.com/golfacademy.hk", "ig": "https://instagram.com/golfpro.hk", "web": "https://golfacademy.hk"},
    {"fb": None, "ig": None, "web": None},  # None
    {"fb": "https://facebook.com/dancestudio.hk", "ig": None, "web": None},
    {"fb": None, "ig": "https://instagram.com/swimcoach.david", "web": None},
    {"fb": None, "ig": None, "web": "https://boxinggym.com"},
    {"fb": "https://facebook.com/tabletennishk", "ig": "https://instagram.com/ttcoach.hk", "web": None},
    {"fb": "https://facebook.com/taekwondohk", "ig": "https://instagram.com/tkdmaster.hk", "web": "https://tkdacademy.hk"},
    {"fb": "https://facebook.com/gymnasticshk", "ig": None, "web": None},
    {"fb": None, "ig": "https://instagram.com/fencingcoach.hk", "web": "https://fencingpro.com"},
    {"fb": "https://facebook.com/volleyballhk", "ig": "https://instagram.com/vbcoach.hk", "web": None},
    {"fb": None, "ig": None, "web": "https://runnersclub.hk"},
    {"fb": "https://facebook.com/danceacademy.hk", "ig": "https://instagram.com/dancepro.hk", "web": "https://dancestudio.hk"},
    {"fb": "https://facebook.com/fitnesshk", "ig": None, "web": "https://personaltrainer.hk"},
    {"fb": None, "ig": "https://instagram.com/badmintoncoach", "web": None},
]

def update_coach_social_media():
    print("🔄 Updating coach social media links...\n")
    
    # Create SSL context that doesn't verify certificates (needed for Supabase)
    ssl_context = ssl.create_default_context()
    ssl_context.check_hostname = False
    ssl_context.verify_mode = ssl.CERT_NONE
    
    conn = None
    try:
        # Connect to database
        conn = psycopg2.connect(
            DATABASE_URL,
            sslmode='require',
            sslrootcert=None
        )
        
        print("✅ Connected to database\n")
        
        cursor = conn.cursor(cursor_factory=RealDictCursor)
        
        # Get all coaches
        cursor.execute("SELECT id, name FROM coaches ORDER BY id")
        coaches = cursor.fetchall()
        
        print(f"Found {len(coaches)} coaches to update\n")
        
        stats = {"fb": 0, "ig": 0, "web": 0, "none": 0}
        
        for i, coach in enumerate(coaches):
            combo = social_combos[i % len(social_combos)]
            
            cursor.execute(
                """
                UPDATE coaches 
                SET facebook_url = %s, instagram_url = %s, website_url = %s 
                WHERE id = %s
                """,
                (combo["fb"], combo["ig"], combo["web"], coach["id"])
            )
            
            # Stats
            has_any = combo["fb"] or combo["ig"] or combo["web"]
            if not has_any:
                stats["none"] += 1
            if combo["fb"]:
                stats["fb"] += 1
            if combo["ig"]:
                stats["ig"] += 1
            if combo["web"]:
                stats["web"] += 1
            
            social_text = []
            if combo["fb"]:
                social_text.append("FB")
            if combo["ig"]:
                social_text.append("IG")
            if combo["web"]:
                social_text.append("Web")
            if not social_text:
                social_text.append("(none)")
            
            print(f"✅ {coach['name']}: {' '.join(social_text)}")
        
        # Commit changes
        conn.commit()
        
        print("\n📊 Summary:")
        print(f"   Facebook: {stats['fb']} coaches")
        print(f"   Instagram: {stats['ig']} coaches")
        print(f"   Website: {stats['web']} coaches")
        print(f"   No social: {stats['none']} coaches")
        print("\n✨ Done! Refresh localhost:5173 to see the changes.")
        
    except psycopg2.Error as e:
        print(f"❌ Database error: {e}")
        if conn:
            conn.rollback()
        raise
    except Exception as e:
        print(f"❌ Error: {e}")
        raise
    finally:
        if conn:
            conn.close()

if __name__ == "__main__":
    try:
        update_coach_social_media()
    except Exception as e:
        print(f"\n❌ Script failed: {e}")
        import traceback
        traceback.print_exc()
        exit(1)

File: scripts/update-social-media.mjs
// Update existing coaches with random social media links - simplified version
import { createRequire } from "module";
const require = createRequire(import.meta.url);
const postgres = require("../lib/db/node_modules/postgres");

const DATABASE_URL = "postgresql://postgres.vosbykmmrfcyrdrkvinx:Felixisagenius!@aws-1-ap-southeast-2.pooler.supabase.com:5432/postgres";

const sql = postgres(DATABASE_URL, { ssl: { rejectUnauthorized: false }, prepare: false });

// Social media link templates
const fbTemplates = [
  (name) => `https://facebook.com/coach.${name}`,
  (name) => `https://facebook.com/${name}.fitness`,
  (name) => `https://facebook.com/${name}.sports`,
];

const igTemplates = [
  (name) => `https://instagram.com/coach_${name}`,
  (name) => `https://instagram.com/${name}_fit`,
  (name) => `https://instagram.com/${name}_hk`,
];

const webTemplates = [
  (name) => `https://${name}.com`,
  (name) => `https://${name}-sports.hk`,
];

function randomItem(arr) {
  return arr[Math.floor(Math.random() * arr.length)];
}

function toUsername(name) {
  return name.toLowerCase().replace(/\s+/g, "").replace(/[^a-z0-9]/gi, "");
}

async function main() {
  console.log("Fetching coaches...");
  const coaches = await sql`SELECT id, name FROM coaches`;
  console.log(`Found ${coaches.length} coaches`);

  for (const coach of coaches) {
    const username = toUsername(coach.name);
    
    // Randomly decide which social media to include
    // Some coaches have IG only, some have FB+IG, some have all three, some have none
    const rand = Math.random();
    
    let facebookUrl = null;
    let instagramUrl = null;
    let websiteUrl = null;

    if (rand < 0.15) {
      // 15% - No social media (all null)
      console.log(`${coach.name}: No social media`);
    } else if (rand < 0.35) {
      // 20% - Instagram only
      instagramUrl = randomItem(igTemplates)(username);
      console.log(`${coach.name}: IG only`);
    } else if (rand < 0.55) {
      // 20% - Facebook only  
      facebookUrl = randomItem(fbTemplates)(username);
      console.log(`${coach.name}: FB only`);
    } else if (rand < 0.75) {
      // 20% - FB + IG
      facebookUrl = randomItem(fbTemplates)(username);
      instagramUrl = randomItem(igTemplates)(username);
      console.log(`${coach.name}: FB + IG`);
    } else if (rand < 0.90) {
      // 15% - IG + Website
      instagramUrl = randomItem(igTemplates)(username);
      websiteUrl = randomItem(webTemplates)(username);
      console.log(`${coach.name}: IG + Website`);
    } else {
      // 10% - All three
      facebookUrl = randomItem(fbTemplates)(username);
      instagramUrl = randomItem(igTemplates)(username);
      websiteUrl = randomItem(webTemplates)(username);
      console.log(`${coach.name}: All three!`);
    }

    await sql`
      UPDATE coaches 
      SET facebook_url = ${facebookUrl}, 
          instagram_url = ${instagramUrl}, 
          website_url = ${websiteUrl}
      WHERE id = ${coach.id}
    `;
  }

  console.log("\n✅ All coaches updated!");
  await sql.end();
}

main().catch(console.error);
